nix/default.nix

```diff
@@ -2,20 +2,21 @@
 let
   nix-pre-commit-hooks = (import (builtins.fetchTarball "https://github.com/cachix/pre-commit-hooks.nix/tarball/master/" + "/nix/") { sources = sources; }).packages;
   overlay = _self: pkgs:
-    let sharedOverrides = {
-        overrides = _self: super: {
-            mkDerivation = args: super.mkDerivation (args //
-                {
-                    # skip running tests for Hackage packages
-                    doCheck = args.pname == "ghcide" || args.pname == "haskell-language-server";
-                    # relax upper bounds
-                    jailbreak = args.pname != "jailbreak-cabal";
-                });
-            };
-            };
-        gitignoreSource = (import sources.gitignore { inherit (pkgs) lib; }).gitignoreSource;
-        extended = haskellPackages:
-          haskellPackages.extend (pkgs.haskell.lib.packageSourceOverrides {
+    let
+        sharedOverrides = {
+          overrides = _self: super: {
+            mkDerivation = args:
+              let
+                imBroken = with builtins;
+                  if hasAttr args.pname super then
+                    super.${args.pname}.meta.broken
+                  else
+                    false;
+              in super.mkDerivation (args // { jailbreak = imBroken; });
```

Comment:
That override is so beautiful. Maybe throw a `{ doCheck = !imBroken }` in there, too?

Reply:
Than you could get rid of the manual hiedb override again. Oh, man that solution is sooo good, I’m hyped.

Reply:
> Maybe throw a { doCheck = !imBroken } in there, too?

Hmm, I was going to do that, but it finally turned out:
```
error: --- EvalError -------------------------------------------------------------------------------------------------------------------------- nix-shell
at: (12:21) in file: /home/berberman/Desktop/haskell/haskell-language-server/nix/default.nix

    11|                   if hasAttr args.pname super then
    12|                     super.${args.pname}.meta.broken
      |                     ^
    13|                   else

infinite recursion encountered
(use '--show-trace' to show detailed location information)

```
I have no idea :(

> That override is so beautiful.

Agree! It works as expected on linux: https://github.com/haskell/haskell-language-server/runs/2083888202, but on macOS, there are still too many things to rebuild: https://github.com/haskell/haskell-language-server/runs/2083888222. Perhaps I missed something?


And similar to f8c741a (https://github.com/haskell/haskell-language-server/runs/2067577443), the build on macOS run into:

```
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [../../../common/unix.mk:61: Tracy-release] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-tracy-0.7.6.drv-0/source/profiler/build/unix'
make: *** [Makefile:7: release] Error 2
make: Leaving directory '/private/tmp/nix-build-tracy-0.7.6.drv-0/source/profiler/build/unix'
builder for '/nix/store/gczrzdw35pw0idnpxl20qpsmmaxcbml0-tracy-0.7.6.drv' failed with exit code 2
error: build of '/nix/store/fi1rkc67njzpcivf69r9avp6da053wp6-ghc-8.10.4-with-packages.drv', '/nix/store/gczrzdw35pw0idnpxl20qpsmmaxcbml0-tracy-0.7.6.drv' failed
```

It's a non-haskell package https://github.com/NixOS/nixpkgs/blob/nixpkgs-unstable/pkgs/development/tools/tracy/default.nix we add in shell
https://github.com/haskell/haskell-language-server/blob/05f25c9d57b50cb7425173cec0b1af4c7581a764/shell.nix#L54

Maybe this package from nixpkgs-unstable is broken on macOS?

Reply:
What about
```
-                imBroken = with builtins;
-                  if hasAttr args.pname super then
-                    super.${args.pname}.meta.broken
-                  else
-                    false;
-              in super.mkDerivation (args // { jailbreak = imBroken; });
-            hiedb = pkgs.haskell.lib.dontCheck super.hiedb;
+                broken = args.broken or false;
+              in super.mkDerivation (args // { jailbreak = broken; doCheck = !broken; });
```
that could fix the first problem. Not sure about the macOs thing, but probably we just need a nixpkgs commit where it’s not broken.

Reply:
Thanks, this solution is more concise and powerful. How to confirm that `tracy` is really broken? Is it possible to reproduce the build elsewhere?

Reply:
Yep, it’s broken.

https://hydra.nixos.org/job/nixpkgs/trunk/tracy.x86_64-darwin